alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -0,0 +1,509 @@
+
+		$childJobs = $batchJob->getRootJob()->getChildJobs();
+		/**
+		 * check if all child jobs are finished(all clip jobs and then a single concat job)
+		 * @var BatchJob $job */
+		foreach ($childJobs as $job)
+		{
+			if ($job->getStatus() != BatchJob::BATCHJOB_STATUS_FINISHED )
+			{
+				KalturaLog::info("number of children:   ". count($childJobs));
+				KalturaLog::info('Child job id [' . $job->getId() . '] status [' . $job->getStatus() . ']');
+				return false;
+			}
+		}
+		if ($batchJob->getJobType() == BatchJobType::POSTCONVERT  &&
+			$batchJob->getStatus() == BatchJob::BATCHJOB_STATUS_FINISHED)
+		{
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * @param $assets
+	 * @return array
+	 * @throws kCoreException
+	 * @throws Exception
+	 */
+	private function getFilesPath($assets): array
+	{
+		$files = array();
+		foreach ($assets as $asset) {
+			/**
+			 * Don't take source it is empty
+			 * @var flavorAsset $asset */
+			if ($asset->getIsOriginal()) {
+				continue;
+			}
+			$syncKey = $asset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
+			$fileSync = kFileSyncUtils::getReadyFileSyncForKey($syncKey);
+			//fileSync should be local
+			if (!$fileSync[1])
+			{
+				throw new kCoreException("Clip Does not exist locally operations stops, there will be no concat");
+			}
+			/*** @var array $fileSync */
+			if ($fileSync[0]->getFullPath())
+			{
+				$files[] = $fileSync[0]->getFullPath();
+			}
+		}
+		return $files;
+	}
+
+	/**
+	 * @param entry $tempEntry
+	 * @return flavorAsset
+	 * @throws kCoreException
+	 */
+	private function getNewAssetFromEntry($tempEntry): flavorAsset
+	{
+
+		/** @var flavorAsset $flavorAsset */
+		$flavorAsset =  assetPeer::getNewAsset(assetType::FLAVOR);
+		// create asset
+		$flavorAsset->setPartnerId($tempEntry->getPartnerId());
+		$flavorAsset->setEntryId($tempEntry->getId());
+		$flavorAsset->setStatus(asset::ASSET_STATUS_QUEUED);
+		$flavorAsset->setFlavorParamsId(kClipAttributes::SYSTEM_DEFAULT_FLAVOR_PARAMS_ID);
+		$flavorAsset->setIsOriginal(false);
+		$flavorAsset->save();
+		return $flavorAsset;
+	}
+
+
+	/**
+	 * @param $partnerId
+	 * @return entry
+	 * @throws Exception
+	 */
+	public function createTempEntryForClip($partnerId)
+	{
+		$tempEntry = new entry();
+		$tempEntry->setType(entryType::MEDIA_CLIP);
+		$tempEntry->setMediaType(entry::ENTRY_MEDIA_TYPE_VIDEO);
+		$tempEntry->setName('TEMP_'.time());
+		$tempEntry->setPartnerId($partnerId);
+		$tempEntry->setStatus(entryStatus::NO_CONTENT);
+		$tempEntry->setDisplayInSearch(EntryDisplayInSearchType::SYSTEM);
+		$tempEntry->setSourceType(EntrySourceType::CLIP);
+		$tempEntry->setConversionProfileId(myPartnerUtils::getConversionProfile2ForPartner($partnerId)->getId());
+		$tempEntry->save();
+		return $tempEntry;
+	}
+
+	/**
+	 * @param string $entryId
+	 * @param  FileSyncKey $concatSyncKey
+	 * @throws Exception
+	 * @throws KalturaAPIException
+	 */
+	private function addDestinationEntryAsset($entryId, $concatSyncKey)
```

Comment:
This function better be divided to 2 functions, one flow of newAssets and another flow with existing asset that has mediaInfo.